### PR TITLE
Instance creation FloatingIPs from novaclient to neutronclient

### PR DESCRIPTION
**Pull request by junousi**
Novaclient floatingIP functionality seems to be gone in queens.

### CODE REVIEW

File: files/nrpe/check_openstack.py [hunk 1]
```diff
@@ -322,10 +343,18 @@ def delete_orphaned_instances(self):
...
-      self.nova.floating_ips.delete(tenant_ip)
-    if len(self.nova.floating_ips.list()) != 0:
-      logging.warn('All floating IPs of instance creation test tenant were not deleted.')
+    # This used to be done with novaclient where it was enought to just list().
+    # With neutronclient one needs to be much more specific, otherwise if ran
+    # with an admin user there is a risk of getting all the Floating IPs.
+    own_project_id = self.session.get_project_id()
+    fip_lookup_params = { 'project_id': own_project_id, }
+    for project_ip in self.neutron.list_floatingips(**fip_lookup_params)['floatingips']:
+      if project_ip['project_id'] != own_project_id:
```
Comment (junousi) [inline files/nrpe/check_openstack.py:352]: In https://github.com/CSCfi/puppet-opsviewagent/pull/70 it was discussed this still needs extra check for admin

File: files/nrpe/check_openstack.py [hunk 2]
```diff
@@ -322,14 +344,39 @@ def delete_orphaned_instances(self):
...
+    # have to care whether the admin role comes from user or group assignment.
+    # If that fails, (we get an exception within this function), we *do not* raise
+    # that particular exception. It's a bit wonky and if we'd refactor the code a bit
+    # more and/or use return values it could probably be more readable.
+    assignments = None
+    try:
+      assignments = self.keystone.role_assignments.list()
+    except:
+      pass
+    assert assignments == None
```
Comment (Aegil) [inline files/nrpe/check_openstack.py:374]: It's not clear how this works. It gets a list of assignments but shouldnt they have member / etc.

Comment (junousi) [reply files/nrpe/check_openstack.py:374]: `keystone.role_assignments.list()` will yield a 401 if `self.session` is not an admin session. The idea is to use that as a simple condition for determening the admin-ness of the calling user.

One thing to understand and is that while a quick-and-dirty like `openstack role assignment list --names --role admin --project foobar|grep admin` is possible to do in very minimal fashion, doing the same thing with python API is a bit more involved (at least to yours truly). One might need to explicitly pass the domain to capture all possible corner cases:

- admin role assigned to user in service domain with some linked project (example: rally)
- admin role assigned to user in service domain without any project (example: heat_admin)
- admin role assigned to user in Default domain (example: heat)
- admin role assigned to group in Default domain (example: nagiostest)
- admin role assigned to group in service domain (this is not a thing in cPouta but should the code still take it into account)

If as reviewer you want this kind of implementation, would it be ok to limit the lookup to the Default domain?

**Comment by Aegil** [comment]
This looks a bit safer. Approving

**Pull request merged**